scipy/stats/_stats_py.py

```diff
@@ -9710,14 +9719,62 @@ def wasserstein_distance(u_values, v_values, u_weights=None, v_weights=None):
     :math:`\mathbb{R} \times \mathbb{R}` whose marginals are :math:`u` and
     :math:`v` on the first and second factors respectively.
 
-    If :math:`U` and :math:`V` are the respective CDFs of :math:`u` and
-    :math:`v`, this distance also equals to:
+    In the 1-dimensional case, let :math:`U` and :math:`V` denote the
+    respective CDFs of :math:`u` and :math:`v`, this distance also equals to:
 
     .. math::
 
         l_1(u, v) = \int_{-\infty}^{+\infty} |U-V|
 
-    See [2]_ for a proof of the equivalence of both definitions.
+    See [3]_ for a proof of the equivalence of both definitions.
+
+    In the more general (higher dimensional) and discrete case, it is also
+    called the optimal transport problem or the Monge problem, which can be
+    expressed in this form,
+
+    Given :math:`\Gamma` the transport plan, :math:`D` the distance matrix and
+    follows,
+
+    .. math::
+
+        x = vec(\Gamma)
+        c = vec(D)
+        b = \begin{bmatrix}
+                u\\
+                v\\
+            \end{bmatrix}
+
+    The Monge problem can be tranformed into a linear programming problem by
+    taking :math:`A x = b` as constraints and :math:`z = c^T x` as minimization
+    target (sum of costs) , where matrix :math:`A` has the form
+
+    .. math::
+
+        \begin{array} {rrrr|rrrr|r|rrrr}
+            1 & 1 & \dots & 1 & 0 & 0 & \dots & 0 & \dots & 0 & 0 & \dots &
+                0 \cr
+            0 & 0 & \dots & 0 & 1 & 1 & \dots & 1 & \dots & 0 & 0 &\dots &
+                0 \cr
+            \vdots & \vdots & \ddots & \vdots & \vdots & \vdots & \ddots
+                & \vdots & \vdots & \vdots & \vdots & \ddots & \vdots  \cr
+            0 & 0 & \dots & 0 & 0 & 0 & \dots & 0 & \dots & 1 & 1 & \dots &
+                1 \cr \hline
+
+            1 & 0 & \dots & 0 & 1 & 0 & \dots & \dots & \dots & 1 & 0 & \dots &
+                0 \cr
+            0 & 1 & \dots & 0 & 0 & 1 & \dots & \dots & \dots & 0 & 1 & \dots &
+                0 \cr
+            \vdots & \vdots & \ddots & \vdots & \vdots & \vdots & \ddots &
+                \vdots & \vdots & \vdots & \vdots & \ddots & \vdots \cr
+            0 & 0 & \dots & 1 & 0 & 0 & \dots & 1 & \dots & 0 & 0 & \dots & 1
+        \end{array}
+
+    By solving the dual form of the above linear programming problem (with
+    solution :math:`y^*`), the Wasserstein distance :math:`l_1 (u, v)` can
+    be computed as :math:`b^T x^*`.
+
+    The above solution is inspired by Vincent Herrmann's blog [5].For a
```

Comment:
```suggestion
    The above solution is inspired by Vincent Herrmann's blog [5]_. For a
```

Reply:
Thanks!

Reply:
Sure thing. Please let me know when you think you have addressed all comments and this is ready for another look.

Reply:
Thanks a lot! I think now it should be all clean and you can have look at the comments now, and if you have extra time could you also check the `TestWassersteinDistance` class (mostly the last two methods)? I have updated those two tests with `pytest` but don't know if it meets your standards :smile: .